Bottom: 3b97057e7d97d88a22328d9e0767329a7b99bb02
Top:    2cc0405c7d1f652fe83eed3caec7d2b923af056b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-01 20:11:39 +0000

Workaround crash where contour is passed as NULL to poly_PreContour









---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
